### PR TITLE
Added support for multiple alternative messages/callsigns

### PR DESCRIPTION
**Pull request by tekk**
As per #45 
- moved `button_tx` to separate `button` config subitem and added `alt_message` option (both set to true by default)
- added support for multiple callsigns / messages / icons / overlays (switch by holding the middle button)
- made `beacons` JSON config to be an array (you can add as much beacon configs as you want)

### CODE REVIEW

**Comment by peterus** [comment]
@tekk I like the idea and will definitely merge it. I will take some more time to analyze the changes as I think there are some small improvements.

**Comment by tekk** [comment]
That's awesome @peterus thanks for letting me know.
If there is anything that I can fix straight away, please let me know and I will do my best to improve the code.
I also see that the `pio check` and formatting checks are failing. (I don't know why?)
Obviously I tested the functionality before making a PR and also guys from our local club are testing my branch on their T-Beams.
If there is anything that I could help with, please let me know.

File: src/LoRa_APRS_Tracker.cpp [hunk 1]
```diff
@@ -178,14 +187,16 @@ void loop() {
 
   if (send_update && gps_loc_update) {
     send_update         = false;
-    nextBeaconTimeStamp = now() + (Config.smart_beacon.active ? Config.smart_beacon.slow_rate : (Config.beacon.timeout * SECS_PER_MIN));
+    Configuration::Beacon beacon = Config.GetCurrentBeacon();
```
Comment (peterus) [inline src/LoRa_APRS_Tracker.cpp:190]: Do not get the current beacon config all the time when the message should be sent. The beacon config will not change so often, instead use the `handle_next_beacon` function to store a pointer to the config.

File: src/LoRa_APRS_Tracker.cpp [hunk 2]
```diff
@@ -40,6 +40,11 @@ static void handle_tx_click() {
   send_update = true;
 }
 
+static void handle_next_beacon() {
+  Configuration::Beacon beacon = Config.SetNextBeacon();
```
Comment (peterus) [inline src/LoRa_APRS_Tracker.cpp:44]: store the beacon config local in the module and use it in the different functions.

**Review by peterus — CHANGES_REQUESTED** [review]
Hi @tekk!
I found some improvements which should be fixed.
If you have questions, please get in contact with me again.
Don't care about the failing style-checker, I will fix it when you are finished.

**Comment by peterus** [comment]
@tekk do you need some help to fix all the requested changes? :)

**Comment by tekk** [comment]
@peterus Yes I indeed need help. I started to rewrite the code, but I'm not that much familiar with C++. Could you please help fixing the issues?

**Comment by tekk** [comment]
@peterus It would be best if you could do the fixes as it's years ago when I last worked with C++. I'm not familiar with smart pointers either. May I please ask you to fix my code? Apart from that, I'm also very busy with my daily job. Please help me fix this branch so we could all profit 😄 Apologies for the dirty code.

**Comment by tekk** [comment]
I was thinking of something like this to manage the current Beacon:

```
class BeaconManager {
  public:
    BeaconManager(Config &config) {};
    std::list<Config::Beacon>::iterator currentBeacon;
    const Configuration::Beacon* const nextBeacon() const;

  private:
    const Configuration::Config* const config;
}
```

But I'm not entirely sure how to implement it.
@peterus Please help!

**Comment by peterus** [comment]
thanks @tekk for the initial idea!

**Pull request merged**